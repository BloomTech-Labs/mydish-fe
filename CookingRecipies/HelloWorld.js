import React from 'react';
import {Text, View} from 'react-native';

const HelloWorld = ({name}) => {
    const message = `Hello ${name}`;

    return (
        <View>
            <Text testID="hw">{message}</Text>)
        </View>
    )
}

export default HelloWorld;

