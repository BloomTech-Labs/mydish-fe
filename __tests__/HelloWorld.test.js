import React from "react";
import HelloWorld from "../HelloWorld";
import { render } from "react-native-testing-library";
import renderer from "react-test-renderer";

describe("HelloWorld", () => {
    it("snapshot hello world", () => {
        const helloSnap = renderer.create(<HelloWorld />).toJSON();
        expect(helloSnap).toMatchSnapshot();
    });

    it("renders the correct message", () => {
        const { getByTestId } = render(<HelloWorld />);
        const text = getByTestId("hw")
        expect(text.props.children).toMatch(/Hello /i);
    });

    it("display correct passed in name", () => {
        const { queryByText } = render(<HelloWorld name="Josh" />);
        const existing = queryByText("Hello Josh");
        expect(existing).not.toBeNull();
    });
});
