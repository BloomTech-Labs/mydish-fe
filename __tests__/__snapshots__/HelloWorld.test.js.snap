// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld snapshot hello world 1`] = `
<View>
  <Text
    testID="hw"
  >
    Hello undefined
  </Text>
</View>
`;
